site/app/models/gradeable/Gradeable.php

```diff
@@ -353,6 +378,26 @@ public function __construct(Core $core, array $details) {
             $this->setPeerBlind($details['peer_blind']);
         }
 
+        if (array_key_exists('autograding', $details)) {
+            $this->setPeerAutograding($details['autograding'] !== null ? $details['autograding'] : true);
+        }
+
+        if (array_key_exists('rubric', $details)) {
+            $this->setPeerRubric($details['rubric'] !== null ? $details['rubric'] : true);
+        }
+
+        if (array_key_exists('files', $details)) {
+            $this->setPeerFiles($details['files'] !== null ? $details['files'] : true);
+        }
+
+        if (array_key_exists('solution_notes', $details)) {
+            $this->setPeerSolutions($details['solution_notes'] !== null ? $details['solution_notes'] : true);
+        }
+
+        if (array_key_exists('discussion', $details)) {
+            $this->setPeerDiscussion($details['discussion'] !== null ? $details['discussion'] : true);
+        }
```

Comment:
```suggestion
$mapping = [
    'autograding' => 'setPeerAutograding',
    'rubric' => 'setPeerRubric',
    'files' => 'setPeerFiles',
    'solution_notes' => 'setPeerSolutions',
    'discussion' => 'setPeerDiscussion'
];

foreach ($mapping as $key => $method) {
    if (array_key_exists($key, $details)) {
        $this->$method($details[$key] ?? true);
    }
}

```

Reply:
Can we do something like this to make it less repetitive? I think we can even apply this to surrounding code.  Maybe in new PR for refactor.

Reply:
sure @ziesski , it removes redundancy
let me commit suggestion right here.